### DISCUSSION TITLE
Extract plain text from PDF file

### DISCUSSION THREAD

**Discussion by JasperAlexander** [wojtekmaj/react-pdf#1411] [Q&A] [answered]
I want to extract the plain text out of a PDF file. Is this possible with react-pdf? If so, how?

**Comment by wojtekmaj** [comment, marked as answer]
Sure, it's possible! `page` that you get in onLoadSuccess callback in Page component has `.getTextContent()` method that does just that. If you need JUST the text though, you can use pdfjs-dist package directly :)

**Reply by githubdebo** [reply]
getting getTextContent() undefined, how to get the page object

**Reply by wojtekmaj** [reply]
@githubdebo Please provide an example of what you're doing now and we'll see what might be the problem.

**Reply by JacksonPalaka** [reply]
Hey,i found a way to do so

**Reply by JacksonPalaka** [reply]
"I've found a way to extract text from PDFs without the need for any additional dependencies or libraries. 
It's a standalone solution!
I hope i was able to solve your querry : )
Here is the link to the solution : https://www.codewithsundeep.com/2023/03/extract-pdf-text-javascript.html
